### ISSUE TITLE
Mag58 Gas Setting

### ISSUE THREAD

**Issue by beno83au** [discoADFU/Project_Uncut_Issues#37]
In reality the Mag58 has an tonne of gas settings and it would be nice to see a slower ROF added to it's fire mode in-game to represent the ability to fire slowly (and sound amazing).

**Comment by MontyBravo** [comment]
The current plan, is to revisit the weapon settings to have alternate fire modes, both a 'safty' fire mode (for those non ACE users) as well as the additional gas settings for the Minimi, Maximi and MAG 58.

**Comment by Azza276** [comment]
In reality the Ton of Gas settings are there to "Balance" the weapon, so the bolt carrier only moves back far enough to effectively cycle the gun, without hitting the harder rear buffer which causes excessive recoil. Yes it affects the rate of fire, but the normal RoF is specified within tolerance. Granted it could be investigated if the RoF is correct or not though. Edit: Investigated 750-1000 Rounds Per Minute.

**Comment by Azza276** [comment]
The Gas settings for Minimi and Maximi are there to compensate for heavy weapon usage where carbon is starting to build up, slowing the RoF or causing a mis-fire due to the weapon not cycling enough rearward... @MontyBravo unless you are interested in scripting in the number of rounds fired, causing a stoppage, necessitating the need for a Gas change (from normal to adverse) I do not believe there is a need for this.

**Comment by beno83au** [comment]
Having fired an enormous amount of rounds through these things (as any digger in a DFSW PL would) i understand the reason for the gas settings. Although i would agree that simulating them all would be pretty excessive, i still think it would be beneficial to include a second, slower rate of fire.

I agree with the point on the F89s though. I guess it depends on how realistic they want to get with them. I'm not 100% on how heat levels are determined on barrels in-game for thermal signatures but i would assume something similar could be done to stimulate a build up of carbon.

**Comment by AtreusTFD** [comment]
Gas settings for F89 min and max are always set to adverse now as standard least in our unit. There are also two types of gas plugs if we are getting anal about it. One with a thumb tab and one without.

On with what beno was saying, Doesn't one of the weapons from Apex have two types of firing mods in it with a faster and slower rate of fire could possibly copy that for the same gas effect.

**Comment by Azza276** [comment]
Yes, Atreus, that is correct, they are set on adverse only now, for all F89's. All the ones with the tab should be replaced with the knurled ones eventually.

**Comment by beno83au** [comment]
Atreus, it could be one of the DLC guns. I remembered something from either a dlc, a mod or a vbs gun being that way. TBH, the first thing i did with uncut was put a dude with a Mag58 in to have a look. 

On the LSW gas though (and i may be remembering this wrong) but i think guys were coming out of kapooka maybe 4ish years ago having being taught to use adverse gas only (among other weird and wonderful things, like the grenade gas setting on the F88 was for firing the GLA......). These last 2 comments just reminded me of it, good point though.

**Comment by Azza276** [comment]
Yeah I had a refresher on F89 about a year ago and they had the new gas reg set to always adverse, and I remember it was a pain to remove the reg without the tab... for others... me no worries ;)

F88 Grenade setting for GLA... Yeah not likely. Grenade gas setting just a remnant from the the original AUG, for barrel launched grenades, which I don't think the Oz Army ever had... It's likely still part of the Licence Thales/ADI have to make the weapons, where the parts probably still need to be interchangeable with other AUGs.

 @beno83au , Last time I touched the Mag-58 was like 12 years ago, and yeah was told that the Gas setting was set by the armourers for a certain cyclic rate, but advanced users could change as desired... Sorry I doubted you, wasn't aware you were Infantry. :)

**Comment by beno83au** [comment]
;) All good mate, just throwing in some content to give weight to my comment, and i rarely put effort into conveying emotion through messages. I try to keep it to the point, which sometimes ends up sounding a bit arrogant i guess. 

We changed the gas as we saw fit on the sfmg line. But it took an armourer to be able to give me a decent explanation about head spacing and all that. Even to the point that when you rotate the carry handle down you are supposed to count the clicks, and that indicated how often head space needed to be checked. 

During a GLA lesson i was asked why i didn't bother to change the gas setting to grenade. Response from me:
"The fuck would i do that for?" 
I followed with an explanation of how a rifle mortar/grenade works. Those guys used to come out with some epic questions.
